lib/scraper.js

```diff
@@ -99,6 +99,10 @@ Scraper.prototype.getResourceHandler = function getHandler (resource) {
 Scraper.prototype.loadResource = function loadResource (resource) {
 	var self = this;
 
+	if(!self.options.urlFilter(resource.url)){
+		return Promise.resolve(null);
```

Comment:
I don't think that would be right. Rejecting a promise is the same as throwing an exception, and should only be used for exception flows.
Also, if you would do it, you would still need to add `.catch` in the file-handlers.

Btw, did you know that the following are the same?
``` javascript
return doSomethingWhichReturnsAPromise()
  .then(function(item){
    return Promise.resolve(item.property);
  });

return doSomethingWhichReturnsAPromise()
  .then(function(item){
    return item.property;
  });
// Inside a then you should just return a value, if the logic is synchronous. 
// You only need to return a promise if it is asynchronous.
```
And this:
``` javascript
return doSomethingWhichReturnsAPromise()
  .then(function(item){
    return Promise.reject(new Error('Something is wrong!'));
  });

return doSomethingWhichReturnsAPromise()
  .then(function(item){
    throw new Error('Something is wrong!');
  });
// Inside a then you should just throw an error, if the logic is synchronous. 
// You only need to return a rejected promise if it is asynchronous.
```